### PR TITLE
Set things up so that we can publish this as a separate package.

### PR DESCRIPTION
**Pull request by codeviking**
This sets things up to be published to NPM. Here's what
a dry-run looks like:

```
~ npm publish --dry-runnpm notice
npm notice 📦  @allenai/varnish-react-router@0.1.0
npm notice === Tarball Contents ===
npm notice 67B   components/index.js
npm notice 2.3kB package.json
npm notice 206B  components/Link.jsx
npm notice 457B  components/ScrollToTopOnPageChange.jsx
npm notice 1.7kB README.md
npm notice 67B   components/index.d.ts
npm notice 183B  components/Link.d.ts
npm notice 458B  components/ScrollToTopOnPageChange.d.ts
npm notice === Tarball Details ===
npm notice name:          @allenai/varnish-react-router
npm notice version:       0.1.0
npm notice package size:  2.3 kB
npm notice unpacked size: 5.4 kB
npm notice shasum:        675746bc3aa6f692b8b36dadfba883b351fd9b49
npm notice integrity:     sha512-cQPVBl3UGTKv5[...]R0MfsIfjt55ew==
npm notice total files:   8
npm notice
+ @allenai/varnish-react-router@0.1.0
```

A nice, neat `2.3 kB` on the wire and `5.4 kB` on disk.

There's a lot of changes here:

- I renamed `InternalLink` to `Link`. This matches the semantics
  used in `react-router`.
- Separate the published `components/` from the application we use
  for development which now lives in `src/`.
- Create separate TypeScript configuration files for local development
  and building for NPM (the default).
- Simplify local development by reducing it to a single build target
  for that purpose.
- Modify the `webpack` configuration so that the local development
  mechanism works (it was building for me, but couldn't actually be
  used in a browser).

I haven't actually ran `npm publish` yet (I've only tested with
`--dry-run`). I'll do that tomorrow and workout any kinks with
Jon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,69 +1,58 @@
...
-Then use it like so:
+Then you can use it like so:
 
 ```tsx
 import { ThemeProvider } from '@allenai/varnish';
-import { InternalLink } from '@allenai/varnish-react-router';
+import { Link } from '@allenai/varnish-react-router';
 import '@allenai/varnish/dist/varnish.css';
 
+// TODO: We should have a better example (or link to external documentation).
```
Comment (jonborchardt) [inline README.md:20]: yes, i will document usage in varnish and link to it here

File: package.json [hunk 2]
```diff
@@ -2,16 +2,15 @@
...
   "scripts": {
-    "lint": "eslint '**/*.{js,ts,tsx,json}' && echo 'Lint complete.'",
-    "lint:fix": "eslint '**/*.{js,ts,tsx,json}' --fix && echo 'Lint --fix complete.'",
-    "clean": "rm dist/bundle.js",
-    "build-dev": "webpack -d --mode development",
-    "build-prod": "webpack -p --mode production",
-    "start": "webpack-dev-server --hot --mode development",
-    "publish": "npm publish --dry-run"
+    "lint": "eslint '**/*.{ts,tsx,json}' && echo 'Lint complete.'",
+    "lint:fix": "eslint '**/*.{ts,tsx,json}' --fix && echo 'Lint --fix complete.'",
```
Comment (jonborchardt) [inline package.json:10]: lint:fix fails for me, ill look into

File: package.json [hunk 3]
```diff
@@ -28,105 +27,39 @@
...
-  "browserslist": [
-    ">0.2%",
-    "not dead",
-    "not ie <= 11",
-    "not op_mini all"
-  ],
   "devDependencies": {
-    "@allenai/varnish": "^0.9.8",
-    "@hot-loader/react-dom": "^16.13.0+4.12.20",
+    "@allenai/eslint-config-varnish": "^0.0.1",
```
Comment (jonborchardt) [inline package.json:31]: oops, thanks
this means that wherever i copied some code from is doing this wrong.... ill look into it

File: package.json (hunk 2 above)
Comment (codeviking) [reply package.json:10]: That's weird. It works for me. Did you run `yarn` once to install the new dependencies?

File: package.json (hunk 3 above)
Comment (codeviking) [reply package.json:31]: No worries.  It complains about some peer dependencies, so it's probably due for an upgrade (`@alleai/eslint-config-varnish`). We probably just need to update the `eslint` it uses and depends on.

File: package.json (hunk 2 above)
Comment (jonborchardt) [reply package.json]: fixed now

**Pull request merged**